File: catalog/backends/python-django/patterns/memory_leak_service.py
from django.http import JsonResponse

# Pattern: Profiling — gradual memory allocation in a cache
# Adapt: use a domain-relevant cache concept
_cache = {}


def add_to_cache(request):
    key = request.GET.get("key", "default")
    _cache[key] = bytearray(1024 * 100)  # 100KB per entry
    return JsonResponse({"cached": len(_cache), "key": key})

File: catalog/backends/ruby-rails/patterns/memory_leak_service.rb
# Pattern: Profiling — gradual memory allocation in a cache
# Adapt: use a domain-relevant cache concept
CACHE = {}

def self.add_to_cache(key)
  CACHE[key] = "\0" * (1024 * 100) # 100KB per entry
  { cached: CACHE.size, key: key }
end
